src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java

```diff
@@ -424,34 +424,50 @@ protected void createService(String appId, AppDeploymentRequest request, Map<Str
 
 		spec.withSelector(idMap).addNewPortLike(servicePort).endPort();
 
-		Map<String, String> annotations = getServiceAnnotations(request.getDeploymentProperties());
+		Map<String, String> annotations = getServiceAnnotations(request);
 
 		client.services().inNamespace(client.getNamespace()).createNew().withNewMetadata().withName(appId)
 			.withLabels(idMap).withAnnotations(annotations).addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE)
 			.endMetadata().withSpec(spec.build()).done();
 	}
 
-	/**
-	 * Get the service annotations for the deployment request.
-	 *
-	 * @param properties The deployment request deployment properties.
-	 * @return map of annottaions
-	 */
-	protected Map<String, String> getServiceAnnotations(Map<String, String> properties) {
-		Map<String, String> annotations = new HashMap<>();
+	private Map<String, String> getPodAnnotations(AppDeploymentRequest request) {
+		String annotationsProperty = request.getDeploymentProperties()
+				.getOrDefault("spring.cloud.deployer.kubernetes.podAnnotations", "");
 
-		String annotationsProperty = properties.getOrDefault("spring.cloud.deployer.kubernetes.serviceAnnotations", "");
+		if (StringUtils.isEmpty(annotationsProperty)) {
+			annotationsProperty = properties.getPodAnnotations();
+		}
+
+		return getAnnotations(annotationsProperty);
+	}
+
+	private Map<String, String> getServiceAnnotations(AppDeploymentRequest request) {
+		String annotationsProperty = request.getDeploymentProperties()
+				.getOrDefault("spring.cloud.deployer.kubernetes.serviceAnnotations", "");
 
 		if (StringUtils.isEmpty(annotationsProperty)) {
-			annotationsProperty = this.properties.getServiceAnnotations();
+			annotationsProperty = properties.getServiceAnnotations();
 		}
 
-		if (StringUtils.hasText(annotationsProperty)) {
-			String[] annotationPairs = annotationsProperty.split(",");
+		return getAnnotations(annotationsProperty);
+	}
+
+	/**
+	 * Extracts annotations from the provided value
+	 *
+	 * @param annotation The deployment request annotations
+	 * @return {@link Map} of annotations
+	 */
+	private Map<String, String> getAnnotations(String annotation) {
+		Map<String, String> annotations = new HashMap<>();
+
+		if (StringUtils.hasText(annotation)) {
+			String[] annotationPairs = annotation.split(",");
```

Comment:
I guess we could add some unit tests on how this comma separated annotations are retrieved as Map.

Reply:
Perhaps we could conveniently add that as part of the existing integration test. I can add it when merging.